staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/draplugin.go

```diff
@@ -354,7 +354,9 @@ func Start(ctx context.Context, nodeServer interface{}, opts ...Option) (result
 
 		// d.resourceSliceController is set concurrently.
 		d.mutex.Lock()
-		d.resourceSliceController.Stop()
+		if d.resourceSliceController != nil {
+			d.resourceSliceController.Stop()
```

Comment:
I found that it is necessary to check whether it is nil, otherwise there will be potential panic

Reply:
Why is this nil check not sufficient?

https://github.com/kubernetes/kubernetes/blob/a7fcc89ac0e7124158eee58820bbf517e0a15377/staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go#L142-L144

Reply:
But not a demo actually using this particular controller, right?

It's true that in general one cannot assume that calling a method with a pointer receiver is safe to do with a nil pointer. It depends on the implementation of that method. The resource slice controller was specifically written so that the caller doesn't need to check for nil.

Reply:
> But not a demo actually using this particular controller, right?

right. I use demo like this: 
```go
type Controller struct {
	ctx    context.Context
	cancel context.CancelFunc
	wg     sync.WaitGroup
}

func (c *Controller) Stop() {
	if c == nil {
		return
	}
	c.cancel()
	c.wg.Wait()
}

func TestController(t *testing.T) {
	var nilController *Controller
	nilController.Stop()
	fmt.Println("Nil Controller stop called")
}

```

>It's true that in general one cannot assume that calling a method with a pointer receiver is safe to do with a nil pointer. It depends on the implementation of that method. The resource slice controller was specifically written so that the caller doesn't need to check for nil.

Putting boundary conditions in the internal implementation helps reduce the usage cost for callers 👍